public/app/core/context/GrafanaContext.ts

```diff
@@ -31,5 +30,13 @@ export function useGrafana(): GrafanaContextType {
 // @grafana/runtime
 export function useReturnToPreviousInternal() {
   const { chrome } = useGrafana();
-  return chrome.setReturnToPrevious;
+  return useCallback(
+    (title: string, href?: string) => {
+      chrome.setReturnToPrevious({
+        title: title,
+        href: href || locationService.getLocation().pathname,
+      });
+    },
+    [chrome]
+  );
```

Comment:
still think this should include query params as well
```suggestion
  const { pathname, search } = locationService.getLocation();
  return useCallback(
    (title: string, href?: string) => {
      chrome.setReturnToPrevious({
        title: title,
        href: href ?? (pathname + search),
      });
    },
    [chrome]
  );
```

also should we use `??` instead of `||` since i think maybe `''` would be a valid href

Reply:
In which use case is the empty string valid `href`? 🤔

Reply:
the base route maybe? honestly idk if that's valid, but either way ?? better matches your intention. you want to use the passed href unless it's not provided, which is exactly what ?? is for